### PR TITLE
[animations] handle open container dismissing currentState of null;

### PR DESCRIPTION
**Pull request by gadfly361**
When using `OpenContainer`, I am seeing an exception in the following scenario:

- A "Closed" page (i.e., made from closedBuilder) has a list of items. 
- Click on a list item to "Open" a container for that item (i.e., made from openBuilder).
- Something causes an update to the "Closed" page and the list restructures itself. Note, this happens while the "Open" container is still open (in my case, I am polling to refresh the list of items' content)
- Dismiss the "Open" container to transition back to the "Closed" page.
- Get the following exceptions:

```
════════ Exception caught by widgets library ═══════════════════════════════════════════════════════
The following NoSuchMethodError was thrown building AnimatedBuilder(animation: AnimationController#f162b(◀ 1.000; for _OpenContainerRoute)➩ProxyAnimation, dirty, state: _AnimatedState#66875):
The getter 'isInTree' was called on null.
Receiver: null
Tried calling: isInTree
```

```
════════ (35) Exception caught by animation library ════════════════════════════════════════════════
The setter 'placeholderSize=' was called on null.
Receiver: null
Tried calling: placeholderSize=null
```

This PR is to handle the above exceptions, by handling the cases when dismissing an open container yields a currentState of null.

### CODE REVIEW

File: packages/animations/lib/src/open_container.dart [hunk 1]
```diff
@@ -287,6 +287,7 @@ class _HideableState extends State<_Hideable> {
   /// When non-null the child is replaced by a [SizedBox] of the set size.
   Size get placeholderSize => _placeholderSize;
   Size _placeholderSize;
+
```
Comment (goderbauer) [inline packages/animations/lib/src/open_container.dart]: Per flutter's style guide there shouldn't be a space here. Please remove.

File: packages/animations/lib/src/open_container.dart [hunk 2]
```diff
@@ -302,6 +303,7 @@ class _HideableState extends State<_Hideable> {
   /// (i.e. [isInTree] returns false).
   bool get isVisible => _visible;
   bool _visible = true;
+
```
Comment (goderbauer) [inline packages/animations/lib/src/open_container.dart]: Per flutter's style guide there shouldn't be a space here. Please remove.

**Comment by gadfly361** [comment]
@goderbauer Thanks for the quick review! I will add a test soon and let you know when it is ready 👍

**Comment by gadfly361** [comment]
@goderbauer 
A test to cover this has been added. Please let me know if there is anything else you need!

Here is the test output for this PR:

```
matthew@ ~/gadfly/forks/packages/packages/animations (open-container-handle-null-on-dismiss) $ flutter test test/open_container_test.dart 
00:03 +19: All tests passed!
```

And here is the test output when running the test against master:
```
matthew@ ~/gadfly/forks/packages/packages/animations (master) $ flutter test test/open_container_test.dart 
00:03 +18: Container can be dismissed after container widget itself is removed without crash                  
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following NoSuchMethodError was thrown building AnimatedBuilder(animation:
AnimationController#cd62e(◀ 1.000; for _OpenContainerRoute)➩ProxyAnimation, dirty, state:
_AnimatedState#e919e):
The getter 'isInTree' was called on null.
Receiver: null
Tried calling: isInTree

The relevant error-causing widget was:
  AnimatedBuilder
  file:///Users/matthew/gadfly/forks/packages/packages/animations/lib/src/open_container.dart:653:14

When the exception was thrown, this was the stack:
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:53:5)
#1      _OpenContainerRoute.buildPage.<anonymous closure> (package:animations/src/open_container.dart:738:63)
#2      AnimatedBuilder.build (package:flutter/src/widgets/transitions.dart:1012:12)
#3      _AnimatedState.build (package:flutter/src/widgets/transitions.dart:178:48)
#4      StatefulElement.build (package:flutter/src/widgets/framework.dart:4334:27)
#5      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4223:15)
#6      Element.rebuild (package:flutter/src/widgets/framework.dart:3947:5)
#7      BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2432:33)
#8      AutomatedTestWidgetsFlutterBinding.drawFrame (package:flutter_test/src/binding.dart:974:18)
#9      RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:283:5)
#10     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1102:15)
#11     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1041:9)
#12     AutomatedTestWidgetsFlutterBinding.pump.<anonymous closure> (package:flutter_test/src/binding.dart:872:9)
#15     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:69:41)
#16     AutomatedTestWidgetsFlutterBinding.pump (package:flutter_test/src/binding.dart:859:27)
#17     WidgetTester.pumpAndSettle.<anonymous closure> (package:flutter_test/src/widget_tester.dart:424:23)
#20     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:69:41)
#21     WidgetTester.pumpAndSettle (package:flutter_test/src/widget_tester.dart:419:27)
#22     main.<anonymous closure> (file:///Users/matthew/gadfly/forks/packages/packages/animations/test/open_container_test.dart:1476:26)
<asynchronous suspension>
#23     testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:124:25)
#24     TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:696:19)
<asynchronous suspension>
#27     TestWidgetsFlutterBinding._runTest (package:flutter_test/src/binding.dart:679:14)
#28     AutomatedTestWidgetsFlutterBinding.runTest.<anonymous closure> (package:flutter_test/src/binding.dart:1050:24)
#34     AutomatedTestWidgetsFlutterBinding.runTest (package:flutter_test/src/binding.dart:1047:15)
#35     testWidgets.<anonymous closure> (package:flutter_test/src/widget_tester.dart:121:22)
#36     Declarer.test.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:171:27)
<asynchronous suspension>
#37     Invoker.waitForOutstandingCallbacks.<anonymous closure> (package:test_api/src/backend/invoker.dart:242:15)
#42     Invoker.waitForOutstandingCallbacks (package:test_api/src/backend/invoker.dart:239:5)
#43     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:169:33)
#48     Declarer.test.<anonymous closure> (package:test_api/src/backend/declarer.dart:168:13)
#49     Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart:392:25)
#63     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:384:19)
#64     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:418:5)
#65     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:174:12)
(elided 32 frames from class _FakeAsync, package dart:async, package dart:async-patch, and package stack_trace)

════════════════════════════════════════════════════════════════════════════════════════════════════
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following NoSuchMethodError was thrown building AnimatedBuilder(animation:
AnimationController#cd62e(◀ 0.667; for _OpenContainerRoute)➩ProxyAnimation, dirty, state:
_AnimatedState#e919e):
The getter 'isInTree' was called on null.
Receiver: null
Tried calling: isInTree

The relevant error-causing widget was:
  AnimatedBuilder
  file:///Users/matthew/gadfly/forks/packages/packages/animations/lib/src/open_container.dart:653:14

When the exception was thrown, this was the stack:
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:53:5)
#1      _OpenContainerRoute.buildPage.<anonymous closure> (package:animations/src/open_container.dart:738:63)
#2      AnimatedBuilder.build (package:flutter/src/widgets/transitions.dart:1012:12)
#3      _AnimatedState.build (package:flutter/src/widgets/transitions.dart:178:48)
#4      StatefulElement.build (package:flutter/src/widgets/framework.dart:4334:27)
#5      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4223:15)
#6      Element.rebuild (package:flutter/src/widgets/framework.dart:3947:5)
#7      BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2432:33)
#8      AutomatedTestWidgetsFlutterBinding.drawFrame (package:flutter_test/src/binding.dart:974:18)
#9      RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:283:5)
#10     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1102:15)
#11     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1041:9)
#12     AutomatedTestWidgetsFlutterBinding.pump.<anonymous closure> (package:flutter_test/src/binding.dart:872:9)
#15     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:69:41)
#16     AutomatedTestWidgetsFlutterBinding.pump (package:flutter_test/src/binding.dart:859:27)
#17     WidgetTester.pumpAndSettle.<anonymous closure> (package:flutter_test/src/widget_tester.dart:424:23)
<asynchronous suspension>
#20     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:69:41)
#21     WidgetTester.pumpAndSettle (package:flutter_test/src/widget_tester.dart:419:27)
#22     main.<anonymous closure> (file:///Users/matthew/gadfly/forks/packages/packages/animations/test/open_container_test.dart:1476:26)
<asynchronous suspension>
#23     testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:124:25)
#24     TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:696:19)
<asynchronous suspension>
#27     TestWidgetsFlutterBinding._runTest (package:flutter_test/src/binding.dart:679:14)
#28     AutomatedTestWidgetsFlutterBinding.runTest.<anonymous closure> (package:flutter_test/src/binding.dart:1050:24)
#34     AutomatedTestWidgetsFlutterBinding.runTest (package:flutter_test/src/binding.dart:1047:15)
#35     testWidgets.<anonymous closure> (package:flutter_test/src/widget_tester.dart:121:22)
#36     Declarer.test.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:171:27)
<asynchronous suspension>
#37     Invoker.waitForOutstandingCallbacks.<anonymous closure> (package:test_api/src/backend/invoker.dart:242:15)
#42     Invoker.waitForOutstandingCallbacks (package:test_api/src/backend/invoker.dart:239:5)
#43     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:169:33)
#48     Declarer.test.<anonymous closure> (package:test_api/src/backend/declarer.dart:168:13)
#49     Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart:392:25)
#63     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:384:19)
#64     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:418:5)
#65     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:174:12)
(elided 32 frames from class _FakeAsync, package dart:async, package dart:async-patch, and package stack_trace)

════════════════════════════════════════════════════════════════════════════════════════════════════
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following NoSuchMethodError was thrown building AnimatedBuilder(animation:
AnimationController#cd62e(◀ 0.333; for _OpenContainerRoute)➩ProxyAnimation, dirty, state:
_AnimatedState#e919e):
The getter 'isInTree' was called on null.
Receiver: null
Tried calling: isInTree

The relevant error-causing widget was:
  AnimatedBuilder
  file:///Users/matthew/gadfly/forks/packages/packages/animations/lib/src/open_container.dart:653:14

When the exception was thrown, this was the stack:
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:53:5)
#1      _OpenContainerRoute.buildPage.<anonymous closure> (package:animations/src/open_container.dart:738:63)
#2      AnimatedBuilder.build (package:flutter/src/widgets/transitions.dart:1012:12)
#3      _AnimatedState.build (package:flutter/src/widgets/transitions.dart:178:48)
#4      StatefulElement.build (package:flutter/src/widgets/framework.dart:4334:27)
#5      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4223:15)
#6      Element.rebuild (package:flutter/src/widgets/framework.dart:3947:5)
#7      BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2432:33)
#8      AutomatedTestWidgetsFlutterBinding.drawFrame (package:flutter_test/src/binding.dart:974:18)
#9      RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:283:5)
#10     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1102:15)
#11     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1041:9)
#12     AutomatedTestWidgetsFlutterBinding.pump.<anonymous closure> (package:flutter_test/src/binding.dart:872:9)
#15     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:69:41)
#16     AutomatedTestWidgetsFlutterBinding.pump (package:flutter_test/src/binding.dart:859:27)
#17     WidgetTester.pumpAndSettle.<anonymous closure> (package:flutter_test/src/widget_tester.dart:424:23)
<asynchronous suspension>
#20     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:69:41)
#21     WidgetTester.pumpAndSettle (package:flutter_test/src/widget_tester.dart:419:27)
#22     main.<anonymous closure> (file:///Users/matthew/gadfly/forks/packages/packages/animations/test/open_container_test.dart:1476:26)
<asynchronous suspension>
#23     testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:124:25)
#24     TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:696:19)
<asynchronous suspension>
#27     TestWidgetsFlutterBinding._runTest (package:flutter_test/src/binding.dart:679:14)
#28     AutomatedTestWidgetsFlutterBinding.runTest.<anonymous closure> (package:flutter_test/src/binding.dart:1050:24)
#34     AutomatedTestWidgetsFlutterBinding.runTest (package:flutter_test/src/binding.dart:1047:15)
#35     testWidgets.<anonymous closure> (package:flutter_test/src/widget_tester.dart:121:22)
#36     Declarer.test.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:171:27)
<asynchronous suspension>
#37     Invoker.waitForOutstandingCallbacks.<anonymous closure> (package:test_api/src/backend/invoker.dart:242:15)
#42     Invoker.waitForOutstandingCallbacks (package:test_api/src/backend/invoker.dart:239:5)
#43     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:169:33)
#48     Declarer.test.<anonymous closure> (package:test_api/src/backend/declarer.dart:168:13)
#49     Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart:392:25)
#63     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:384:19)
#64     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:418:5)
#65     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:174:12)
(elided 32 frames from class _FakeAsync, package dart:async, package dart:async-patch, and package stack_trace)

════════════════════════════════════════════════════════════════════════════════════════════════════
══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
The following NoSuchMethodError was thrown building AnimatedBuilder(animation:
AnimationController#cd62e(◀ 0.000; for _OpenContainerRoute)➩ProxyAnimation, dirty, state:
_AnimatedState#e919e):
The getter 'isInTree' was called on null.
Receiver: null
Tried calling: isInTree

The relevant error-causing widget was:
  AnimatedBuilder
  file:///Users/matthew/gadfly/forks/packages/packages/animations/lib/src/open_container.dart:653:14

When the exception was thrown, this was the stack:
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:53:5)
#1      _OpenContainerRoute.buildPage.<anonymous closure> (package:animations/src/open_container.dart:738:63)
#2      AnimatedBuilder.build (package:flutter/src/widgets/transitions.dart:1012:12)
#3      _AnimatedState.build (package:flutter/src/widgets/transitions.dart:178:48)
#4      StatefulElement.build (package:flutter/src/widgets/framework.dart:4334:27)
#5      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4223:15)
#6      Element.rebuild (package:flutter/src/widgets/framework.dart:3947:5)
#7      BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2432:33)
#8      AutomatedTestWidgetsFlutterBinding.drawFrame (package:flutter_test/src/binding.dart:974:18)
#9      RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:283:5)
#10     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1102:15)
#11     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1041:9)
#12     AutomatedTestWidgetsFlutterBinding.pump.<anonymous closure> (package:flutter_test/src/binding.dart:872:9)
#15     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:69:41)
#16     AutomatedTestWidgetsFlutterBinding.pump (package:flutter_test/src/binding.dart:859:27)
#17     WidgetTester.pumpAndSettle.<anonymous closure> (package:flutter_test/src/widget_tester.dart:424:23)
<asynchronous suspension>
#20     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:69:41)
#21     WidgetTester.pumpAndSettle (package:flutter_test/src/widget_tester.dart:419:27)
#22     main.<anonymous closure> (file:///Users/matthew/gadfly/forks/packages/packages/animations/test/open_container_test.dart:1476:26)
<asynchronous suspension>
#23     testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:124:25)
#24     TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:696:19)
<asynchronous suspension>
#27     TestWidgetsFlutterBinding._runTest (package:flutter_test/src/binding.dart:679:14)
#28     AutomatedTestWidgetsFlutterBinding.runTest.<anonymous closure> (package:flutter_test/src/binding.dart:1050:24)
#34     AutomatedTestWidgetsFlutterBinding.runTest (package:flutter_test/src/binding.dart:1047:15)
#35     testWidgets.<anonymous closure> (package:flutter_test/src/widget_tester.dart:121:22)
#36     Declarer.test.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:171:27)
<asynchronous suspension>
#37     Invoker.waitForOutstandingCallbacks.<anonymous closure> (package:test_api/src/backend/invoker.dart:242:15)
#42     Invoker.waitForOutstandingCallbacks (package:test_api/src/backend/invoker.dart:239:5)
#43     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:169:33)
#48     Declarer.test.<anonymous closure> (package:test_api/src/backend/declarer.dart:168:13)
#49     Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart:392:25)
#63     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:384:19)
#64     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:418:5)
#65     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:174:12)
(elided 32 frames from class _FakeAsync, package dart:async, package dart:async-patch, and package stack_trace)

════════════════════════════════════════════════════════════════════════════════════════════════════
══╡ EXCEPTION CAUGHT BY ANIMATION LIBRARY ╞═════════════════════════════════════════════════════════
The following NoSuchMethodError was thrown while notifying status listeners for ProxyAnimation:
The setter 'placeholderSize=' was called on null.
Receiver: null
Tried calling: placeholderSize=null

When the exception was thrown, this was the stack:
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:53:5)
#1      _OpenContainerRoute.didPush.<anonymous closure> (package:animations/src/open_container.dart:545:15)
#2      AnimationLocalStatusListenersMixin.notifyStatusListeners (package:flutter/src/animation/listener_helpers.dart:193:19)
#3      AnimationLocalStatusListenersMixin.notifyStatusListeners (package:flutter/src/animation/listener_helpers.dart:193:19)
#4      AnimationController._checkStatusChanged (package:flutter/src/animation/animation_controller.dart:773:7)
#5      AnimationController._tick (package:flutter/src/animation/animation_controller.dart:789:5)
#6      Ticker._tick (package:flutter/src/scheduler/ticker.dart:237:5)
#7      SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1102:15)
#8      SchedulerBinding.handleBeginFrame.<anonymous closure> (package:flutter/src/scheduler/binding.dart:1017:11)
#9      _LinkedHashMapMixin.forEach (dart:collection-patch/compact_hash.dart:379:8)
#10     SchedulerBinding.handleBeginFrame (package:flutter/src/scheduler/binding.dart:1015:17)
#11     AutomatedTestWidgetsFlutterBinding.pump.<anonymous closure> (package:flutter_test/src/binding.dart:868:9)
#14     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:69:41)
#15     AutomatedTestWidgetsFlutterBinding.pump (package:flutter_test/src/binding.dart:859:27)
#16     WidgetTester.pumpAndSettle.<anonymous closure> (package:flutter_test/src/widget_tester.dart:424:23)
<asynchronous suspension>
#19     TestAsyncUtils.guard (package:flutter_test/src/test_async_utils.dart:69:41)
#20     WidgetTester.pumpAndSettle (package:flutter_test/src/widget_tester.dart:419:27)
#21     main.<anonymous closure> (file:///Users/matthew/gadfly/forks/packages/packages/animations/test/open_container_test.dart:1476:26)
<asynchronous suspension>
#22     testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:124:25)
#23     TestWidgetsFlutterBinding._runTestBody (package:flutter_test/src/binding.dart:696:19)
<asynchronous suspension>
#26     TestWidgetsFlutterBinding._runTest (package:flutter_test/src/binding.dart:679:14)
#27     AutomatedTestWidgetsFlutterBinding.runTest.<anonymous closure> (package:flutter_test/src/binding.dart:1050:24)
#33     AutomatedTestWidgetsFlutterBinding.runTest (package:flutter_test/src/binding.dart:1047:15)
#34     testWidgets.<anonymous closure> (package:flutter_test/src/widget_tester.dart:121:22)
#35     Declarer.test.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:171:27)
<asynchronous suspension>
#36     Invoker.waitForOutstandingCallbacks.<anonymous closure> (package:test_api/src/backend/invoker.dart:242:15)
#41     Invoker.waitForOutstandingCallbacks (package:test_api/src/backend/invoker.dart:239:5)
#42     Declarer.test.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/declarer.dart:169:33)
#47     Declarer.test.<anonymous closure> (package:test_api/src/backend/declarer.dart:168:13)
#48     Invoker._onRun.<anonymous closure>.<anonymous closure>.<anonymous closure>.<anonymous closure> (package:test_api/src/backend/invoker.dart:392:25)
#62     _Timer._runTimers (dart:isolate-patch/timer_impl.dart:384:19)
#63     _Timer._handleMessage (dart:isolate-patch/timer_impl.dart:418:5)
#64     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:174:12)
(elided 32 frames from class _FakeAsync, package dart:async, package dart:async-patch, and package stack_trace)

The ProxyAnimation notifying status listeners was:
  AnimationController#cd62e(⏮ 0.000; paused; for _OpenContainerRoute)➩ProxyAnimation
════════════════════════════════════════════════════════════════════════════════════════════════════
══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following message was thrown:
Multiple exceptions (5) were detected during the running of the current test, and at least one was
unexpected.
════════════════════════════════════════════════════════════════════════════════════════════════════
00:03 +18 -1: Container can be dismissed after container widget itself is removed without crash [E]           
  Test failed. See exception logs above.
  The test description was: Container can be dismissed after container widget itself is removed without crash
  
00:03 +18 -1: Some tests failed.
```

**Comment by goderbauer** [comment]
What does the animation look like for this case? Does the container scale down to its old original "closed" location and then just disappear when it has reached its old target size? That would be kinda janky.

**Comment by gadfly361** [comment]
@goderbauer Yeah, that is exactly right - the container scales down to its old original 'closed' location and then disappears once it has reached its old target size.

I think it comes down to trade-offs, and I see two options (if you can think of more, please let me know!)

Option 1
- The container shrinks back to its old location and then disappears
  - Pro: we at least understand the container is being removed
  - Con: visually has a little jank when the container disappears at the end of the transition

Option 2
- There is no transition pertaining to the container when it gets dismissed
  - Pro: visually there is no jank
  - Con: doesn't give feedback to user that the container was removed. I imagine people may misinterpret a different container as being the one that was just opened if it took the removed containers location on the page.

Do you have any thoughts?

**Comment by goderbauer** [comment]
@jojomcbobo do you have any opinion on how the container should close if the original closed container doesn't exist anymore?

**Comment by jonasnaimark** [comment]
I think option 1 makes sense. Ideally the jank would be addressed by adding a second transition after the container shrinks. Either with a fade https://pub.dev/packages/animations#fade, or having the container slide off screen similar to how when you delete an email in Gmail. I'd suggest those second transitions are beyond the scope of the container transform itself.

**Comment by gadfly361** [comment]
@goderbauer @jonasnaimark Sounds good to me 👍  If the secondary transitions are out of scope, is the current state of this PR ok then?

**Comment by goderbauer** [comment]
I am ok with taking this approach then. Can you fix the merge conflict please?

**Comment by gadfly361** [comment]
@goderbauer Fixed the merge conflict and the build checks have passed 👍

**Comment by gadfly361** [comment]
@goderbauer I think this is ready, but let me know if there is anything else that is still needed 👍

**Comment by goderbauer** [comment]
https://cirrus-ci.com/build/4678353963712512 indicates that the analyze check actually passed. Merging.

**Pull request merged**